modules/lib/hyprlang.nix

```diff
@@ -0,0 +1,56 @@
+{ lib }: rec {
+  valueType = with lib.types;
+    nullOr
+    (oneOf [ bool int float str path (attrsOf valueType) (listOf valueType) ])
+    // {
+      description = "Hyprlang configuration value";
+    };
+
+  toHyprconf = with lib;
```

Comment:
Maybe having an argset instead of multiple args would be better, given their number. Also allows us to set defaults.